### PR TITLE
Deprecate the routes only version of the config

### PR DESCRIPTION
**Pull request by cah-briangantzler**

### CODE REVIEW

**Comment by cah-briangantzler** [comment]
So use to have prettier run when I commit things, forgot to run it separately here. 

Mind if I add lint-staged and run prettier then lint on commit. Commits will fail to commit if lint fails, which is good, and prettier will pretty it up so lint wont fail on prettier problems.

File: tests/dummy/app/pods/docs/advanced/server-configuration/template.md [hunk 1]
```diff
@@ -1,22 +1,13 @@
 # Server configuration
 
-The MirageJS server is configured for you by ember-cli-mirage. However, if you
```
Comment (SergeAstapov) [inline tests/dummy/app/pods/docs/advanced/server-configuration/template.md:3]: @cah-briangantzler minor, thinking if would be more helpful to say `The MirageJS server used to be configured for you by ember-cli-mirage.`

File: tests/dummy/app/pods/docs/advanced/server-configuration/template.md [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-`mirage/serializers`, to create your Mirage JS server when your app boots up
-in development and testing.
-
-You can now opt in to having more control over exactly how your Mirage server
-is instantiated using the same code as the mirageJS examples of creating a server 
-by changing your current default export that represents the routes to a normal function,
-then creating a new default export function that creates the mirageJS server.
-
-This new default export function receives a single argument named `config`, 
+The MirageJS server has been configured for you by ember-cli-mirage. This hase been
```
Comment (SergeAstapov) [inline tests/dummy/app/pods/docs/advanced/server-configuration/template.md:3]: "This hase been" -> "This has been"

**Comment by SergeAstapov** [comment]
> So use to have prettier run when I commit things, forgot to run it separately here.
> 
> Mind if I add lint-staged and run prettier then lint on commit. Commits will fail to commit if lint fails, which is good, and prettier will pretty it up so lint wont fail on prettier problems.

sounds good!
Mind to do such setup in separate PR?

**Comment by SergeAstapov** [comment]
@cah-briangantzler CI should be green as of #2327, mind to rebase onto master?

**Comment by cah-briangantzler** [comment]
I thought I was up to date. My bad I didnt check. Was wondering about that error showing up again. Repushed, waiting on checks.

Yea of course a separate PR

**Comment by cah-briangantzler** [comment]
Updated the docs, then rebased onto master.Then remembered, this is a deprecation and shouldnt be on master, it should be on v2 shouldnt it?

I updated the docs for V2. All these docs need re-written again with V3 and there be a migration section from V2 to V3.   


I think I need to rebase this back to V2. Were the checks green on the V2 branch ?

**Comment by cah-briangantzler** [comment]
closing and starting over, history is muddled

**Pull request closed without merging**